osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -405,9 +406,12 @@ private ServiceFunctionChain validateAndLoadSfcBind(EntityManager em, BindSecuri
              VirtualSystem vs = validateAndLoadVirtualSystem(em, virtualSystemId);
              
              //if this virtual system is pointing to more than one SFC and one of the SFC other than given SFC
-             //is already binded to SG and active, throw exception
-            if (vs.getServiceFunctionChains().size() > 1) {
-                for (ServiceFunctionChain serviceFunctionChain : vs.getServiceFunctionChains()) {
+             //is already binded to SG and active, throw exception;
+             //Get All the sfcs having this virutalsystem in the chain
+             List<ServiceFunctionChain> serviceFunctionChainList = ServiceFunctionChainEntityMgr.
+                                                                 listServiceFunctionChainsByVirtualSystem(em, vs);
+            if (serviceFunctionChainList != null && !serviceFunctionChainList.isEmpty()) {
```

Comment:
!isEmpty is not needed. The loop wont happen if empty

Reply:
I wanted to check of list exist and not empty.  will query return just null if no data ?.
please let me know.

Reply:
You can leave the null check but the empty check is not necessary. If empty the loop below will not run at all, it is already safe.

Reply:
removed empty check